torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -414,10 +414,21 @@ def foreach_norm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> TupleStrateg
     return TupleStrategy(output_tuple_strategy_childs)
 
 
-@register_op_strategy([aten._linalg_svd.default], schema_info=RuntimeSchemaInfo(1))
-def linalg_svd_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
-    # Since we do not have a simple way to compute a sharded SVD, always fall
-    # back to replicate
+@register_op_strategy(
+    [
+        aten._linalg_svd.default,
+        aten.linalg_qr.default,
+        # TODO: `diagonal_copy` can have an improved sharding strategy for
+        # shard placements that does not require redistributing to replicate.
+        aten.diagonal_copy.default,
```

Comment:
Should we also add `diag_embed`?
Looks like `diag` could also call into it from here:
https://github.com/pytorch/pytorch/blob/main/torch/_refs/__init__.py#L4260C22-L4260C32